src/btree/bt_npos.c

```diff
@@ -0,0 +1,362 @@
+                    __wt_sleep(0, 10);
+                    goto restart;
+                }
+                /* Fall through */
+            case WT_REF_DELETED:
+                /*
+                 * Can't go down from here. Return the "current" page and
+                 * __find_closest_leaf will finish the job.
+                 */
+                goto done;
+            default: /* WT_REF_DISK, WT_REF_MEM, WT_REF_SPLIT */
+                goto descend;
+            }
+            /* Unreachable but it's ok to be here. */
+        }
+        /* Unreachable but it's ok to be here. */
+
+descend:
+        /*
+         * Swap the current page for the child page. If the page splits while we're retrieving it,
+         * restart the search at the root.
+         *
+         * On other error, simply return, the swap call ensures we're holding nothing on failure.
+         */
+        if ((ret = __wt_page_swap(session, current, descent, flags)) == 0) {
+            current = descent;
+            continue;
+        }
+        if (read_cache && (ret == WT_NOTFOUND || ret == WT_RESTART))
+            goto done;
+        if (ret == WT_RESTART) {
+            WT_RET(__wt_page_release(session, current, flags));
+            goto restart;
+        }
+        return (ret);
+    }
+done:
+    /*
+     * Because eviction considers internal pages in post-order, returning the root page will
+     * indicate the end of walk. Also, eviction will never evict the root. Also, returning a NULL is
+     * not an error for eviction but a signal to start over. So handle this case individually.
+     */
+    if (read_cache && __wt_ref_is_root(current)) {
+        WT_RET(__wt_page_release(session, current, flags));
+        current = NULL;
+    }
+    *refp = current;
+    return (0);
+}
+
+/*
+ * __wt_page_from_npos --
+ *     Find a page given its normalized position.
+ */
+int
+__wt_page_from_npos(
+  WT_SESSION_IMPL *session, WT_REF **refp, double npos, uint32_t read_flags, uint32_t walk_flags)
+{
+    WT_DECL_RET;
+
+    WT_WITH_PAGE_INDEX(session, ret = __page_from_npos_internal(session, refp, npos, read_flags));
+    WT_RET(ret);
+    /* Return the first good page starting from here. */
+    return (__find_closest_leaf(session, refp, walk_flags));
+}
+
+/*
+ * !!!
+ * __wt_page_from_npos_for_eviction --
+ *     Go to a page given its normalized position (for eviction).
+ *     - Use WT_READ_PREV to look up backwards.
+ */
+int
+__wt_page_from_npos_for_eviction(
+  WT_SESSION_IMPL *session, WT_REF **refp, double npos, uint32_t read_flags, uint32_t walk_flags)
+{
+    return (__wt_page_from_npos(session, refp, npos, read_flags | WT_READ_EVICT_READ_FLAGS,
+      walk_flags | WT_READ_EVICT_WALK_FLAGS));
+}
+
+/*
+ * !!!
+ * __wt_page_from_npos_for_read --
+ *     Go to a leaf page given its normalized position (for reading).
+ *     - Use WT_READ_PREV to look up backwards.
+ */
+int
+__wt_page_from_npos_for_read(
```

Comment:
I was thinking about this for a while but generally I am against adding unused functions to the code, which would be:
```
__wt_page_from_npos
__wt_page_from_npos_for_read
```
They add unused and untested complexity and should be avoided. If they are required in the future they can be added then.